sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -32,138 +32,60 @@
-                for header, value in http_request.headers.items():
-                    log_string += "\n    '{}': '{}'".format(header, value)
-                if isinstance(http_request.body, types.GeneratorType):
-                    log_string += "\nFile upload"
-                    logger.info(log_string)
-                    return
-                try:
-                    if isinstance(http_request.body, types.AsyncGeneratorType):
-                        log_string += "\nFile upload"
-                        logger.info(log_string)
-                        return
-                except AttributeError:
-                    pass
-                if http_request.body:
-                    log_string += "\nA body is sent with the request"
-                    logger.info(log_string)
-                    return
-                log_string += "\nNo body was attached to the request"
-                logger.info(log_string)
-
-            except Exception as err:  # pylint: disable=broad-except
-                logger.warning("Failed to log request: %s", repr(err))
-        else:
-            super().on_request(request)
-
-
-    def on_response(self, request, response):  # pylint: disable=too-many-return-statements, too-many-statements
-        # type: (PipelineRequest, PipelineResponse) -> None
-        if self._enable_diagnostics_logging:
-            http_response = response.http_response
-            ir = http_response.internal_response
-            try:
-                logger = response.context["logger"]
-
-                if not logger.isEnabledFor(logging.INFO):
-                    return
-
+            if self._enable_diagnostics_logging:
+                ir = response.http_response.internal_response
                 multi_record = os.environ.get(HttpLoggingPolicy.MULTI_RECORD_LOG, False)
                 if multi_record:
-                    logger.info("Response status: %r", http_response.status_code)
                     logger.info("Response status reason: %r", ir.reason)
                     try:
                         logger.info("Elapsed Time: %r", ir.elapsed)
                     except AttributeError:
                         logger.info("Elapsed Time: %r", None)
                     if http_response.status_code >= 400:
-                        sm = ir.text
-                        sm.replace("true", "True")
-                        sm.replace("false", "False")
-                        temp_sm = ast.literal_eval(sm)
-                        temp_sm['message'] = temp_sm['message'].replace("\r", " ")
-                        logger.into("Response status error message: %r", temp_sm['message'])
-                    logger.info("Response headers:")
-                    for res_header, value in http_response.headers.items():
-                        logger.info("    %r: %r", res_header, value)
+                        logger.into("Response status error message: %r", self._format_error(ir.text))
```

Comment:
I _think_ in the async transport, calling `.text` might be an asynchronous method.
Even though for an error response the data isn't streamed and should be available synchronously - we should be careful about only using sync APIs of the response object.